From 7e6a2ebea4290c9169e22388a7ba75a4825e12ec Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Wed, 22 Aug 2012 09:46:13 +0800
Subject: [PATCH 330/498] ENGR00219871 camera recording can't be stop
 sometimes

if use the default stop_threshold in the tinyalsa, it will cause the SDMA stop
when the overrun happened in the camera recording.
So change the stop_threshold to the buffersize for input.

Signed-off-by: b02247 <b02247@freescale.com>
---
 alsa/tinyalsa_hal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/alsa/tinyalsa_hal.c b/alsa/tinyalsa_hal.c
index 0bf485e..84729f3 100644
--- a/alsa/tinyalsa_hal.c
+++ b/alsa/tinyalsa_hal.c
@@ -940,6 +940,8 @@ static int start_input_stream(struct imx_stream_in *in)
     }
     LOGW("card %d, port %d device %x", card, port, in->device);
 
+    in->config.stop_threshold = in->config.period_size * in->config.period_count;
+
     if(in->device & AUDIO_DEVICE_IN_ANLG_DOCK_MIC) {
         if((int)in->config.rate != adev_get_rate_for_device(adev, AUDIO_DEVICE_IN_ANLG_DOCK_MIC, PCM_IN) ||
            in->config.channels != 1) {
-- 
1.8.0

